src/material/core/option/option.ts

```diff
@@ -285,8 +292,11 @@ export class MatOption<T = any> extends _MatOptionBase<T> {
     changeDetectorRef: ChangeDetectorRef,
     @Optional() @Inject(MAT_OPTION_PARENT_COMPONENT) parent: MatOptionParentComponent,
     @Optional() @Inject(MAT_OPTGROUP) group: MatOptgroup,
+    @Attribute('role') role: string,
```

Comment:
This won't add much maintenance burden on us, but I'm not sure how useful it would be. `mat-option` was written to be used in `mat-select` and `mat-autocomplete` which have a specific interaction pattern already that might not make sense if the options are with a different role, e.g. `checkbox`.

Reply:
I'm hesitant to add this for a case that we don't really support as it can encourage `mat-option` to be used incorrectly in other cases as well.

Reply:
I would not say that whether the role is set to "option" or "checkbox" is correct or incorrect. Instead, it provides more flexibility to the clients of the component to configure the component in a more accessible way. Both options behave the same to sighted users. 

However, you do raise a good point that it can be misused. Perhaps I can have an assertion in the constructor that ensure role is either option or checkbox, and elaborate in the documentation that to use mat-option as a checkbox, you need to make sure it is aligned with other A11y guidelines.

Reply:
Could this be achieved with a custom directive instead? E.g.

```
@Directive({
  selector: 'mat-option',
  host: {
    'role': 'checkbox'
  }
})
class CustomOption {}
```

Reply:
As is discussed offline with crisbeto@, this will be done on client side by introducing a customized directive. Close the PR for now.